packages/app/src/cli/services/dev/urls.ts

```diff
@@ -1,18 +1,21 @@
 import {updateURLsPrompt} from '../../prompts/dev.js'
 import {api, error, output, plugins, store} from '@shopify/cli-kit'
-import {Plugin} from '@oclif/core/lib/interfaces'
+import {Config} from '@oclif/core'
 
 export interface PartnersURLs {
   applicationUrl: string
   redirectUrlWhitelist: string[]
 }
 
-export async function generateURL(pluginList: Plugin[], frontendPort: number): Promise<string> {
-  const tunnelPlugin = await plugins.lookupTunnelPlugin(pluginList)
-  if (!tunnelPlugin) throw new error.Bug('The tunnel could not be found')
-  const url = await tunnelPlugin?.start({port: frontendPort})
+export async function generateURL(config: Config, frontendPort: number): Promise<string> {
+  // For the moment we assume to always have ngrok, this will change in a future PR
+  // and will need to use "getListOfTunnelPlugins" to find the available tunnel plugins
+  const result = await plugins.runTunnelPlugin(config, frontendPort, 'ngrok')
+
+  if (result.error === 'multiple-urls') throw new error.Bug('Multiple tunnel plugins for ngrok found')
+  if (result.error === 'no-urls' || !result.url) throw new error.Bug('Ngrok failed to start the tunnel')
```

Comment:
Are those bugs?

Reply:
Right now they are, because in the current release they should never happen.

Once we add more tunnel plugins and ngrok is not added by default, these won't be bugs